lib/extras.py

```diff
@@ -437,6 +438,150 @@ def callproc(self, procname, vars=None):
+        if slot_type == REPLICATION_LOGICAL:
+            if output_plugin is None:
+                raise psycopg2.ProgrammingError("output_plugin is required for logical replication slot")
+
+            command += "LOGICAL %s" % self.quote_ident(output_plugin)
+
+        elif slot_type == REPLICATION_PHYSICAL:
+            if output_plugin is not None:
+                raise psycopg2.ProgrammingError("output_plugin is not applicable to physical replication")
+
+            command += "PHYSICAL"
+
+        else:
+            raise psycopg2.ProgrammingError("unrecognized replication slot type")
+
+        self.execute(command)
+
+    def drop_replication_slot(self, slot_name):
+        """Drop streaming replication slot."""
+
+        command = "DROP_REPLICATION_SLOT %s" % self.quote_ident(slot_name)
+        self.execute(command)
+
+    def start_replication(self, slot_type, slot_name=None, writer=None, start_lsn=None,
+                          timeline=0, keepalive_interval=10, options=None):
+        """Start and consume replication stream."""
+
+        command = "START_REPLICATION "
+
+        if slot_type == REPLICATION_LOGICAL and slot_name is None:
+            raise psycopg2.ProgrammingError("slot_name is required for logical replication slot")
+
+        if slot_name:
+            command += "SLOT %s " % self.quote_ident(slot_name)
+
+        if slot_type == REPLICATION_LOGICAL:
+            command += "LOGICAL "
+        elif slot_type == REPLICATION_PHYSICAL:
+            command += "PHYSICAL "
+        else:
+            raise psycopg2.ProgrammingError("unrecognized replication slot type")
+
+        if start_lsn is None:
+            start_lsn = '0/0'
+
+        # reparse lsn to catch possible garbage
+        lsn = start_lsn.split('/')
+        command += "%X/%X" % (int(lsn[0], 16), int(lsn[1], 16))
+
+        if timeline != 0:
+            if slot_type == REPLICATION_LOGICAL:
+                raise psycopg2.ProgrammingError("cannot specify timeline for logical replication")
+
+            if timeline < 0:
+                raise psycopg2.ProgrammingError("timeline must be >= 0: %d" % timeline)
+
+            command += " TIMELINE %d" % timeline
+
+        if options:
+            if slot_type == REPLICATION_PHYSICAL:
+                raise psycopg2.ProgrammingError("cannot specify plugin options for physical replication")
+
+            command += " ("
+            for k,v in options.iteritems():
+                if not command.endswith('('):
+                    command += ", "
+                command += "%s %s" % (self.quote_ident(k), _A(str(v)))
+            command += ")"
+
+        return self.start_replication_expert(command, writer=writer,
+                                             keepalive_interval=keepalive_interval)
+
+    def send_feedback_message(self, written_lsn=0, sync_lsn=0, apply_lsn=0, reply_requested=False):
+        return self.send_replication_feedback(written_lsn, sync_lsn, apply_lsn, reply_requested)
```

Comment:
I don't recall at the moment what stopped me from implementing a subclass of cursor type, but that is the reason.  The one with "replication" in the name is defined on the general cursor type, and the other one is just a thin wrapper/alias on the `ReplicationCursor` type.  Definitely implementing a proper subclass would be better approach.